sycl/include/sycl/ext/oneapi/matrix/matrix-hip.hpp

```diff
@@ -0,0 +1,389 @@
+
+//===-------- matrix-hip.hpp - matrix ext impl ---*- C++ -*-------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===-------------------------------------------------------------------=== //
+
+#pragma once
+#include "matrix-unified-utils.hpp"
+#include <sycl/ext/oneapi/bfloat16.hpp>
+
+#if defined(__gfx90a__)
+#define __HIP_PLATFORM_AMD_MFMA__
+#endif
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+namespace matrix {} // namespace matrix
+} // namespace experimental
+
+using matrix_layout = sycl::ext::oneapi::experimental::matrix::layout;
```

Comment:
I think this could cause potential problems in user code, if for example they choose to use `matrix_layout` as a variable name etc?

I think that fully qualified names are recommended.